src/hooks/useFilteredData.js

```diff
@@ -0,0 +1,41 @@
+import { useProducts } from "../context/products-context";
+import {
+  getFilteredByCategory,
+  getFilteredByRating,
+  getFilteredByPriceRange,
+  getFilteredByPriceSort,
+} from "../utils/products-filter-functions";
+
+export function useFilteredData() {
+  const { products } = useProducts();
+  const {
+    productsList,
+    sortByPrice,
+    filterByCategory,
+    filterByRating,
+    filterByPriceRange,
+  } = products;
+  const filteredData = JSON.parse(JSON.stringify(productsList));
```

Comment:
What's the reason of doing this?

Reply:
To deep copy my productsList array to filteredData.